feat(spdx): add per-upload EnableOsselotExport toggle which improve SPDX & ReadMeOSS exports for OSSelot report

Vaibhavsahu2810 opened the pull request:
<!-- SPDX-FileCopyrightText: © Fossology contributors
     SPDX-License-Identifier: GPL-2.0-only
-->

## Description
This pull request introduces a new per‐upload configuration flag, **EnableOsselotExport**, under  
**Browse → Pfile → Conf → Report Configuration → SPDX Report Settings**. When checked for a given upload, all SPDX-based exports (SPDX-TV, SPDX RDF/XML) and the ReadMeOSS agent adopt OSSelot-compatible behavior (see issue #3057).

### Changes

- **Added per-upload OSSelot toggle**  
  ![Enable OSSelot Export checkbox](https://github.com/user-attachments/assets/2bb1927d-e280-46d2-b059-e9b6310f1853)  
  Introduced a third checkbox (“Enable OSSelot Export”).

- **SPDX-TV Export**  
  - In **License Information**, list **every** license (including those previously unprefixed).  
  - Rewrite custom licenses as `LicenseRef-See-file.LICENSE` (no `-fossology-`).  
    ![SPDX-TV before/after](https://github.com/user-attachments/assets/8bb6c601-6d93-43d4-9da2-3227cc8ede3a)  
  - Print each license’s full text. Include any license comments if present.  
    ![Full text and comments in SPDX-TV](https://github.com/user-attachments/assets/e813fc26-8817-4ba7-85f5-f7dbe8c4ae68)

- **SPDX RDF/XML Export**  
  - Apply identical logic: emit all licenses , strip `-fossology-`, prepend `LicenseRef-` when needed, and include full license text.  
    ![SPDX RDF/XML before/after 1](https://github.com/user-attachments/assets/b74a625d-6d83-40c3-bb6d-83a9bab9e493)  
    ![SPDX RDF/XML before/after 2](https://github.com/user-attachments/assets/f671d231-7d1e-4a47-8740-859c7afdfaee)

- **ReadMeOSS Output**  
  - Strip both `LicenseRef-fossology-` and `LicenseRef-` from each license line.   
    ![ReadMeOSS output without prefixes](https://github.com/user-attachments/assets/c3e8e188-395c-499e-82b1-216e4c54c33e)

## How to test

1. **Enable “Osselot Export” for a upload**  
   - Navigate to **Browse → Pfile → Conf → Report Configuration → SPDX Report Settings** for a given upload.  
   - Check **“Enable OSSelot Export”** and click **Save**.

2. **Generate ReadMeOSS**  
   - Verify that license lines appear **without** any `LicenseRef-…` prefix.  

3. **Generate SPDX-TV Export (.spdx)**  
   - Confirm that custom licenses are rewritten as `LicenseRef-<ShortName>` (no `-fossology-`).  
   - Every license’s full text should appear . License comments (if any) should also be included.

4. **Generate SPDX RDF/XML Export (.spdx.rdf)**  
   - verify `LicenseRef-<ShortName>` (no `-fossology-`) for licenses.  
   - Confirm that it contains the **entire** license text.

CC: @shaheemazmalmmd @JanAltenberg

shaheemazmalmmd commented:
> Should i create getSPDXReportConf($uploadId, $key) in src/lib/php/Util which will act as a centralized function and i can use it in both spdx and readmeoss agent instead of just copying it in readmeoss agent to avoid duplication !!

@Vaibhavsahu2810 : you can use the function which is already there in upload dao to get the report conf per upload. It is already been used in spdx.phpfile.

shaheemazmalmmd commented:
@Vaibhavsahu2810 : Please rebase your branch with current master and squash your commits to one.

Vaibhavsahu2810 commented:
> @Vaibhavsahu2810 : Please rebase your branch with current master and squash your commits to one.

Done !! , Also added the fix for indentation issue as recommended by the @JanAltenberg

Vaibhavsahu2810 commented:
Added two more fixes according to the feedback from @JanAltenberg 
- Listens for changes on the "Enable OSSelot Export" checkbox
When it's checked, automatically checks the "Show SPDX license comments" checkbox
- Added info button for "Enable OSSelot Export" checkbox.

Vaibhavsahu2810 commented:
<img width="1212" height="867" alt="image" src="https://github.com/user-attachments/assets/e35c7215-3752-462a-8c54-5875616bc8c4" />
Now added the flag in user setting page

Vaibhavsahu2810 commented:
Also ,After reviewing the existing REST API, I don’t think we need to create any new endpoints. The current API already provides the required functionality

* **Generate report** – `GET /repo/api/v1/reportParameters`

  * Parameters:

    * `uploadId` – ID of the upload to generate a report for
    * `reportFormat` – Format of the report (`spdx2`, `spdx2tv`, `dep5`, `spdx2csv`, etc.)

* **Update configuration** – `PUT /uploads/{id}/conf`

In found out that, the API already supports configuring and generating OSSelot export reports.

JanAltenberg commented:
Thanks @Vaibhavsahu2810 for the work. I've tested the latest changes. I covered SPDXTV, SPDX RDF and ReadmeOSS exports with and without OSSelot export enabled and all export types worked.
Also the per user default settings worked for me.
@shaheemazmalmmd @GMishx: If tests work for you as well and you don't have any additional findings for the code, I am fine with the PR.

shaheemazmalmmd commented:
@Vaibhavsahu2810 : can you please look into the failing testcases ??
